Providers/Profile.swift

```diff
@@ -502,21 +486,6 @@ public class BrowserProfile: Profile {
         return SQLiteLogins(db: self.loginsDB)
     }()
 
-    // This is currently only used within the dispatch_once block in loginsDB, so we don't
-    // have to worry about races giving us two keys. But if this were ever to be used
-    // elsewhere, it'd be unsafe, so we wrap this in a dispatch_once, too.
-    lazy private var loginsKey: String? = {
-        let key = "sqlcipher.key.logins.db"
```

Comment:
In particular, note that getting the logins key can also side-effect generating one; it's important this this not race with itself, hence the comment above.